neqo-http3/src/connection_client.rs

```diff
@@ -4404,6 +4458,30 @@ mod tests {
         assert_eq!(client.cancel_push(0), Err(Error::InvalidStreamId));
     }
 
+    /// We can't keep the connection alive on the basis of a push promise,
+    /// nor do we want to if the push promise is not interesting to the client.
+    /// We do the next best thing, which is keep any push stream alive if the
+    /// client reads from it.
+    #[test]
+    fn push_keep_alive() {
+        let (mut client, mut server, request_stream_id) = connect_and_send_request(true);
+
+        // Promise a push and deliver, but don't close the stream.
+        send_push_promise(&mut server.conn, request_stream_id, 0);
+        let _ = send_push_data(&mut server.conn, 0, false);
+        let dgram = server.conn.process_output(now()).dgram();
+        client.process_input(dgram.unwrap(), now());
+
+        let mut buf = [0; 16];
+        let (read, fin) = client.push_read_data(now(), 0, &mut buf).unwrap();
+        assert!(read < buf.len());
+        assert!(!fin);
+
+        force_idle(&mut client, &mut server);
+        let idle_timeout = ConnectionParameters::default().get_idle_timeout();
+        assert_eq!(client.process_output(now()).callback(), idle_timeout / 2);
```

Comment:
This test is no really testing the push stream. connect_and_send_request(..) will open a stream tha well also need keeping alive. You need to send response for request_stream_id.